src/system/SystemObject.h

```diff
@@ -194,21 +198,135 @@ union ObjectArena
 };
 
 /**
+ *  @class ObjectPoolBase
+ *
+ *  @brief
+ *    This is the abstract base class for Static and Dynamic ObjectPool objects.
+ *
+ *  @note
+ *      This base class can be used to pass pointers to the sized ObjectPoolStatic and ObjectPoolDynamic classes withouth requiring
+ * the class holding the reference to be templated on the pool size.
+ */
+template <typename T>
+class ObjectPoolBase
+{
+public:
+    virtual ~ObjectPoolBase()                                                                                           = default;
+    virtual T * TryCreate()                                                                                             = 0;
+    virtual void Reset()                                                                                                = 0;
+    virtual void GetStatistics(chip::System::Stats::count_t & aNumInUse, chip::System::Stats::count_t & aHighWatermark) = 0;
```

Comment:
So the cost here is more RAM usage and code, right?  And the cost on at least p6 seems pretty high.... (like 5KB of RAM high?).

But it's a few hundred bytes on every platform.  Is that cost worth the additional flexibility we get via this refactor?

Reply:
@cecille fixed this with 366710b4961261ab8ee0e0964746e70b15413559 … I'm looking into why we don't have a new size report (but GitHub runs aren't searchable :angry:)

Reply:
Missing size reports will be fixed for future runs by #11483; unfortunately the build artifacts incorrectly marked stale are gone.